src/ring_theory/I_filtration.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+
+import algebra.direct_sum.ring
+import algebra.direct_sum.module
+import ring_theory.noetherian
+import ring_theory.rees_algebra
+import ring_theory.finiteness
+
+/-!
+
+# `I`-filtrations of modules
+
+This file contains the definitions and basic results around (stable) `I`-filtrations of modules.
+
+-/
+
+
+universes u v
+
+variables {R M : Type u} [comm_ring R] [add_comm_group M] [module R M] (I : ideal R)
+
+open polynomial
+open_locale polynomial big_operators
+
+/-- An `I`-filtration on the module `M` is a sequence of decreasing submodules `N i` such that
+`I • N ≤ I (i + 1)`. Note that we do not require the filtration to start from `⊤`. -/
+@[ext]
+structure I_filtration (M : Type u) [add_comm_group M] [module R M] :=
```

Comment:
What do you think about calling this `ideal.filtration`? Is that name too ambiguous?

With that spelling, you could write `some_ideal.filtration M` instead of `I_filtraction some_ideal M`.